### PR TITLE
Adds support to multiple 'src' paths

### PR DESCRIPTION
**Pull request by omarduarte**
* The 'src' configuration option can now take in either a String or Array of strings
* Added a 'lib' folder in `fixtures` to test multiple 'src' paths
* Refactored most path related code in order to support multiple 'src' paths

### CODE REVIEW

File: test/expected/default-option-dependencies.json [hunk 1]
```diff
@@ -14,6 +18,8 @@
     "src/modules/desktop/**/*.html",
     "src/modules/tablet/**/*.html",
     "src/modules/tv/**/*.html",
+    "lib/watch/apple/**/*.html",
```
Comment (urgoringo) [inline test/expected/default-option-dependencies.json]: does it support case where bower downloads lib into directory like: `lib/tw-common/common/errors` and Angular module name is: `tw.common.errors`? I guess our libs will be following such naming conventions?

Comment (omarduarte) [inline test/expected/default-option-dependencies.json]: hey @urgoringo this is a great question. This gunt task will only collectDependencies that have a 'Module.js' file in its directory. In the case that we try to collect `tw.common.errors` and `lib/common/errors` doesn't have a `Module.js` file in it, then the dependency would not be collected AND NO EXCEPTIONS WOULD BE RAISED. 

So the real question would be if we should throw an error for these cases. I think we should. What do you think?

**Comment by urgoringo** [comment]
Does that mean throwing an exception when a directory of our lib does not contain `Module.js`? I think we can't do that. We can have some directories which are purely for grouping modules but not a module itself also there are directories for `styles`, `views` that don't include any `Module.js`

**Comment by urgoringo** [comment]
I think we can merge this

**Pull request merged**